lib/gui/modules/drive-scanner.js

```diff
@@ -16,86 +16,130 @@
-)
+/**
+ * @summary Time to delay the first scan (in milliseconds)
+ * @description This prevents the drive scans from kicking
+ * off while Etcher is still loading, which would
+ * considerably delay the time until the main window shows
+ * @type {Number}
+ * @constant
+ */
+const DRIVE_SCANNER_FIRST_SCAN_DELAY_MS = 125
 
-/* eslint-enable lodash/prefer-lodash-method */
+/**
+ * @summary DriveScanner class
+ * @class
+ * @extends {EventEmitter}
+ */
+class DriveScanner extends EventEmitter {
+  /**
+   * @summary DriveScanner constructor
+   * @example
+   * const scanner = new DriveScanner()
+   *   .on('error', (error) => { ... })
+   *   .on('drives', (drives) => { ... })
+   */
+  constructor () {
+    super()
+    this.isRunning = false
+    this.timer = null
+    this.errorCount = 0
+    this.run = this.run.bind(this)
+  }
 
-  .flatMap(() => {
-    return Rx.Observable.fromNodeCallback(drivelist.list)()
-  })
+  /**
+   * @summary Start the scanner
+   * @public
+   * @example
+   * scanner.start()
+   */
+  start () {
+    debug('start')
+    if (this.isRunning) {
+      return
+    }
+    this.isRunning = true
+    this.timer = setTimeout(this.run, DRIVE_SCANNER_FIRST_SCAN_DELAY_MS)
+  }
 
-  .map((drives) => {
-    if (settings.get('unsafeMode')) {
-      return drives
+  /**
+   * @summary Kick off a scan & schedule the next run
+   * @private
+   * @example
+   * scanner.run()
+   */
+  run () {
+    if (!this.isRunning) {
+      return
     }
 
-    return _.reject(drives, {
-      system: true
+    DriveScanner.scan((error, drives) => {
+      debug('scan', error || drives)
+      if (error) {
+        this.errorCount += 1
+        this.emit('error', error)
+      } else {
+        this.emit('drives', drives)
+      }
+      if (this.isRunning) {
+        this.timer = setTimeout(this.run, DRIVE_SCANNER_DELAY_MS)
+      }
     })
-  })
-  .pausable(new Rx.Subject())
+  }
 
-/*
- * This service emits the following events:
- *
- * - `drives (Object[])`
- * - `error (Error)`
- *
- * For example:
- *
- * ```
- * driveScanner.on('drives', (drives) => {
- *   console.log(drives);
- * });
- *
- * driveScanner.on('error', (error) => {
- *   throw error;
- * });
- * ```
- */
-availableDrives.subscribe((drives) => {
-  emitter.emit('drives', drives)
-}, (error) => {
-  emitter.emit('error', error)
-})
+  /**
+   * @summary Stop scanning for drives
+   * @public
+   * @example
+   * scanner.stop()
+   */
+  stop () {
+    debug('stop')
+    this.isRunning = false
+    clearTimeout(this.timer)
+  }
 
-/**
- * @summary Start scanning drives
- * @function
- * @public
- *
- * @example
- * driveScanner.start();
- */
-emitter.start = () => {
-  availableDrives.resume()
-}
+  /**
+   * @summary Scan for drives, optionally filtering out system drives
+   * @private
+   * @static
+   * @param {Function} callback - callback(error, drives)
+   * @example
+   * DriveScanner.scan((error, drives) => {
+   *   ...
+   * })
+   */
+  static scan (callback) {
```

Comment:
What about something like this? It gets rid of the timer, and the extra `.bind()`, plus reduces the code size a bit (disclaimer: I didn't actually run it :P)

```diff
diff --git a/lib/gui/modules/drive-scanner.js b/lib/gui/modules/drive-scanner.js
index 2193fe8e..9a72e84d 100644
--- a/lib/gui/modules/drive-scanner.js
+++ b/lib/gui/modules/drive-scanner.js
@@ -18,7 +18,8 @@
 
 const _ = require('lodash')
 const EventEmitter = require('events').EventEmitter
-const drivelist = require('drivelist')
+const Bluebird = require('bluebird');
+const drivelist = Bluebird.promisifyAll(require('drivelist'))
 const settings = require('../models/settings')
 const debug = require('debug')('etcher:drive-scanner')
 
@@ -71,8 +72,9 @@ class DriveScanner extends EventEmitter {
     if (this.isRunning) {
       return
     }
+
     this.isRunning = true
-    this.timer = setTimeout(this.run, DRIVE_SCANNER_FIRST_SCAN_DELAY_MS)
+    Bluebird.delay(DRIVE_SCANNER_FIRST_SCAN_DELAY_MS).then(this.run);
   }
 
   /**
@@ -86,18 +88,16 @@ class DriveScanner extends EventEmitter {
       return
     }
 
-    DriveScanner.scan((error, drives) => {
-      debug('scan', error || drives)
-      if (error) {
-        this.errorCount += 1
-        this.emit('error', error)
-      } else {
-        this.emit('drives', drives)
-      }
+    DriveScanner.scan().then((drives) => {
+      this.emit('drives', drives)
+
       if (this.isRunning) {
-        this.timer = setTimeout(this.run, DRIVE_SCANNER_DELAY_MS)
+        return Bluebird.delay(DRIVE_SCANNER_DELAY_MS).then(this.run);
       }
-    })
+    }).catch((error) => {
+      this.errorCount += 1
+      this.emit('error', error)
+    });
   }
 
   /**
@@ -109,7 +109,6 @@ class DriveScanner extends EventEmitter {
   stop () {
     debug('stop')
     this.isRunning = false
-    clearTimeout(this.timer)
   }
 
   /**
@@ -122,22 +121,15 @@ class DriveScanner extends EventEmitter {
    *   ...
    * })
    */
-  static scan (callback) {
-    drivelist.list((error, drives) => {
-      if (error) {
-        callback(error)
-        return
-      }
-
-      let list = drives
-
+  static scan () {
+    return drivelist.listAsync().then((drives) => {
       if (!settings.get('unsafeMode')) {
-        list = _.reject(drives, {
+        return _.reject(drives, {
           system: true
         })
       }
 
-      callback(null, list)
+      return drives
     })
   }
 }
```